spec/features/user/signup_spec.rb

```diff
@@ -0,0 +1,50 @@
+require 'rails_helper'
+
+RSpec.describe "user" do
+  it "can sign up with email" do
+    visit root_path
+    click_link_or_button('Sign up')
+    expect(current_path).to eq(new_user_path)
+    fill_in("name", with: "Paul")
+    fill_in("email", with: "test@example.com")
+    fill_in("password", with: "test")
+    click_button('Register')
+    expect(User.count).to eq(1)
+  end
+
+  it "can't sign up with an invalid email" do
+    visit root_path
+    click_link_or_button('Sign up')
+    expect(current_path).to eq(new_user_path)
+    fill_in("name", with: "Paul")
+    fill_in("email", with: "hi")
+    fill_in("password", with: "test")
+    click_button('Register')
+    expect(User.count).to eq(0)
+    expect(page).to have_content("Input is invalid, please try again")
+  end
+
+  it "can't sign up without a full name" do
+    visit root_path
+    click_link_or_button('Sign up')
+    expect(current_path).to eq(new_user_path)
+    fill_in("name", with: "")
+    fill_in("email", with: "test@example.com")
+    fill_in("password", with: "test")
+    click_button('Register')
+    expect(User.count).to eq(0)
+    expect(page).to have_content("Input is invalid, please try again")
+  end
+
+  it "can't sign up without a passowrd" do
+    visit root_path
+    click_link_or_button('Sign up')
+    expect(current_path).to eq(new_user_path)
+    fill_in("name", with: "Paul")
+    fill_in("email", with: "test@example.com")
+    fill_in("password", with: "")
```

Comment:
I didn't do much styling as I thought our long term goal was to make this a modal? 

If we aren't going to go that way, I will fix these.